yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -722,24 +596,6 @@ export class P2PClient<T extends P2PClientType = P2PClientType.Full>
 
     this.log.info(`Moving ${txsToMoveToPending.length} mined txs back to pending`);
     await this.txPool.markMinedAsPending(txsToMoveToPending);
-
-    await this.synchedLatestBlockNumber.set(latestBlock);
-    // no need to update block hashes, as they will be updated as new blocks are added
-  }
-
-  private async startServiceIfSynched() {
-    if (
-      this.currentState === P2PClientState.SYNCHING &&
-      (await this.getSyncedLatestBlockNum()) >= this.latestBlockNumberAtStart &&
-      (await this.getSyncedProvenBlockNum()) >= this.provenBlockNumberAtStart
-    ) {
-      this.log.debug(`Synched to blocks at start`);
-      this.setCurrentState(P2PClientState.RUNNING);
-      if (this.syncResolve !== undefined) {
-        this.syncResolve();
-        await this.p2pService.start();
-      }
-    }
```

Comment:
Is this not needed anymore? My understanding was that this function would block the node from 'starting' until all of its components had synched.

Is there a risk that after a restart a node would have components at different heights because they synched at different speeds from the archiver?

Reply:
You're completely correct, myself and lasse have just discussed this. I'm currently bringing this logic back!